packages/bot-skeleton/src/scratch/hooks/contextmenu.js

```diff
@@ -245,6 +245,9 @@ Blockly.ContextMenu.blockDisableStack = function (block) {
  */
 Blockly.ContextMenu.blockEnableStack = function (block) {
     const checkAreSomeDisabled = (block_, disabledArr = []) => {
+        if (block_.restricted_parents && !block_.restricted_parents?.includes(block_.getTopParent()?.type)) {
```

Comment:
that is a nice catch Nijil, thank you. I had the same thought but then i checked the code and haven't found any restricted parent that is not a topBlock. Restricted parents we have now are 'after_purchase', 'during_purchase', 'before_purchase', 'trade_definition'. If for multipliers you don't restrict TP/SL or multipliers block itself to anything except 'trade_options' that code should work. And given that we have the concept of restricted blocks because we have different stages of contract when some data or action may not be available i think we should restrict only to topBlocks.